### DISCUSSION TITLE
rewrite-spring now requires Java 11 via rewrite-micrometer

### DISCUSSION THREAD

**Discussion by gjd6640** [openrewrite/rewrite#4332] [General] [answered]
As a heads-up: Anyone whose JAVA_HOME is set to a Java 8 installation will fail with the below error when they use the current release of rewrite-maven-plugin. You'll need to adjust JAVA_HOME to a modern JVM version to use it.

```
[ERROR] Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:5.36.0:run (default-cli) on project ntp-timed-job-train-exception-monitor: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:5.36.0:run failed: Could not load class org.openrewrite.micrometer.TimerToObservation : java.lang.UnsupportedClassVersionError: org/openrewrite/micrometer/TimerToObservation has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:5.36.0:run (default-cli) on project ntp-timed-job-train-exception-monitor: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:5.36.0:run failed: Could not load class org.openrewrite.micrometer.TimerToObservation : java.lang.UnsupportedClassVersionError: org/openrewrite/micrometer/TimerToObservation has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
 ```

Note that something similar happened previously and is documented in [this discussion thread](https://github.com/openrewrite/rewrite/discussions/4272). In that case, @timtebeek acted by cutting a new release that again supported Java 8. That may be needed again.

Command line used:
```
mvn -e org.openrewrite.maven:rewrite-maven-plugin:run "-Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-spring:RELEASE" "-Drewrite.activeRecipes=com.MyCompany.cve.UpgradeToSpringFramework5NoSpringBoot,com.MyCompany.cve.ApplySafeCVEFixesConditionally,com.MyCompany.recommended.FixBestPractices,com.MyCompany.recommended.UpdateToLatestTooling" "-Drewrite.exportDatatables=true" "-Drewrite.configLocation=target/remediation-plugin/rewrite.yml"
```

**Comment by gjd6640** [comment]
Looks like an alternative to switching to a newer JVM version is to use a prior release of the rewrite-spring library like so:

> mvn -e org.openrewrite.maven:rewrite-maven-plugin:run "-Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-spring:**5.14.0**" blah blah

**Comment by timtebeek** [comment]
Thanks for the quick heads up once again @gjd6640 ; In this case it's rewrite-micrometer that giving us trouble, due to
https://github.com/openrewrite/rewrite-micrometer/blob/37b2aab82712c5cab1da8e1e1917d2da219edb05/build.gradle.kts#L8-L11

That was newly included in the Spring Boot 3.3 upgrades, as it's a requirement to also upgrade Micrometer.
- https://github.com/openrewrite/rewrite-spring/commit/196dc5f133b61eb56f8ab1c3150fffe68d6f2fd6

Perhaps for now the quickest solution would be to remove that dependency rewrite-spring until we figure out a next step.

**Reply by timtebeek** [reply]
- I've logged an issue to track: https://github.com/openrewrite/rewrite-micrometer/issues/12

**Reply by timtebeek** [reply, marked as answer]
I've just released a new patch version of rewrite-spring that should clear out this issue
- https://github.com/openrewrite/rewrite-spring/releases/tag/v5.15.1

**Reply by gjd6640** [reply]
I've verified that this version works with Java 8. Thanks!

**Reply by timtebeek** [reply]
Thanks for the quick call out and verification of the fix!
